src/lib/icon/icon.md

```diff
@@ -80,9 +80,32 @@ match the current theme's colors using the `color` attribute. This can be change
 
 ### Accessibility
 
-If an `aria-label` attribute is set on the `md-icon` element, its value will be used as-is. If not,
-the md-icon component will attempt to set the aria-label value from one of these sources:
-* The `alt` attribute
-* The `fontIcon` input
-* The name of the icon from the `svgIcon` input (not including any namespace)
-* The text content of the component (for ligature icons)
+Similar to an `<img>` element, an icon alone does not convey any useful information for a
+screen-reader user. The user of `<md-icon>` must provide additional information pertaining to how
+the icon is used.
+
+In thinking about accessibility, it is useful to place icon use into one of three categories:
+1. **Decorative**: the icon conveys no real semantic meaning and is purely cosmetic.
+2. **Interactive**: a user will click or otherwise interact with the icon to perform some action.
+3. **Indicator**: the icon is not interactive, but it conveys some information, such as a status.
+
+#### Decorative icons
+When the icon is puely cosmetic and conveys no real semantic meaning, the `<md-icon>` element
+should be marked with `aria-hidden="true"`.
+
+#### Interactive icons
+Icons alone are not interactive elements for screen-reader users; when the user would interact with
+some icon on the page, a more appropriate  element should "own" the interaction:
+* The `<md-icon>` element should be a child of a `<button>` or `<a>` element.
+* The `<md-icon>` element should be marked with `aria-hidden="true"`.
+* The parent `<button>` or `<a>` should either have a meaningful label provided either through
+direct text content, `aria-label`, or `aria-labelledby`.
+
+#### Indicator icons
```

Comment:
to make this easier we could add a simple component like this (it could also automatically add a tooltip):

```html
<md-indicator [message]="msg">
  <md-icon>error</md-icon>
</md-indicator>
```